internal/planner/planner_test.go

```diff
@@ -224,6 +224,42 @@ func TestPlannerHelloWorld(t *testing.T) {
 			note:    "relation unify",
 			queries: []string{`walk(input, [["foo", y], x])`},
 		},
+		{
+			note:    "else conflict-1",
+			queries: []string{`data.p.q`},
+			modules: []string{
+				`package p
+
+				q {
+					false
+				}
+				else = true {
+					true
+				}
+				q = false
+				`,
+			},
```

Comment:
### Before

I don't think what we had before was wrong, but this is how it changed:
```
| *ir.Funcs Funcs (1 funcs)
| | *ir.Func g0.data.p.q (2 params: [0 1], 3 blocks)
| | | *ir.Block Block (1 statements)
| | | | *ir.BlockStmt BlockStmt (2 blocks) &{0 0 0  }
| | | | | *ir.Block Block (5 statements)
| | | | | | *ir.MakeBooleanStmt &{Value:false Target:3 Location:{Index:0 Col:6 Row:4 file:module-0.rego text:false}}
| | | | | | *ir.MakeBooleanStmt &{Value:false Target:4 Location:{Index:0 Col:6 Row:4 file:module-0.rego text:false}}
| | | | | | *ir.NotEqualStmt &{A:3 B:4 Location:{Index:0 Col:6 Row:4 file:module-0.rego text:false}}
| | | | | | *ir.MakeBooleanStmt &{Value:true Target:5 Location:{Index:0 Col:5 Row:3 file:module-0.rego text:q}}
| | | | | | *ir.AssignVarOnceStmt &{Target:2 Source:5 Location:{Index:0 Col:5 Row:3 file:module-0.rego text:q}}
| | | | | *ir.Block Block (6 statements)
| | | | | | *ir.IsUndefinedStmt &{Source:2 Location:{Index:0 Col:5 Row:6 file:module-0.rego text:else = true}}
| | | | | | *ir.MakeBooleanStmt &{Value:true Target:3 Location:{Index:0 Col:6 Row:7 file:module-0.rego text:true}}
| | | | | | *ir.MakeBooleanStmt &{Value:false Target:4 Location:{Index:0 Col:6 Row:7 file:module-0.rego text:true}}
| | | | | | *ir.NotEqualStmt &{A:3 B:4 Location:{Index:0 Col:6 Row:7 file:module-0.rego text:true}}
| | | | | | *ir.MakeBooleanStmt &{Value:true Target:5 Location:{Index:0 Col:5 Row:6 file:module-0.rego text:else = true}}
| | | | | | *ir.AssignVarOnceStmt &{Target:2 Source:5 Location:{Index:0 Col:5 Row:6 file:module-0.rego text:else = true}}
| | | *ir.Block Block (5 statements)
| | | | *ir.MakeBooleanStmt &{Value:true Target:3 Location:{Index:0 Col:9 Row:9 file:module-0.rego text:false}}
| | | | *ir.MakeBooleanStmt &{Value:false Target:4 Location:{Index:0 Col:9 Row:9 file:module-0.rego text:false}}
| | | | *ir.NotEqualStmt &{A:3 B:4 Location:{Index:0 Col:9 Row:9 file:module-0.rego text:false}}
| | | | *ir.MakeBooleanStmt &{Value:false Target:5 Location:{Index:0 Col:5 Row:9 file:module-0.rego text:q = false}}
| | | | *ir.AssignVarOnceStmt &{Target:2 Source:5 Location:{Index:0 Col:5 Row:9 file:module-0.rego text:q = false}}
| | | *ir.Block Block (1 statements)
| | | | *ir.ReturnLocalStmt &{Source:2 Location:{Index:0 Col:5 Row:3 file:module-0.rego text:q}}
```

### After

```
| *ir.Funcs Funcs (1 funcs)
| | *ir.Func g0.data.p.q (2 params: [0 1], 4 blocks)
| | | *ir.Block Block (1 statements)
| | | | *ir.BlockStmt BlockStmt (3 blocks) &{0 0 0  }
| | | | | *ir.Block Block (5 statements)
| | | | | | *ir.MakeBooleanStmt &{Value:false Target:5 Location:{Index:0 Col:6 Row:4 file:module-0.rego text:false}}
| | | | | | *ir.MakeBooleanStmt &{Value:false Target:6 Location:{Index:0 Col:6 Row:4 file:module-0.rego text:false}}
| | | | | | *ir.NotEqualStmt &{A:5 B:6 Location:{Index:0 Col:6 Row:4 file:module-0.rego text:false}}
| | | | | | *ir.MakeBooleanStmt &{Value:true Target:7 Location:{Index:0 Col:5 Row:3 file:module-0.rego text:q}}
| | | | | | *ir.AssignVarStmt &{Source:7 Target:3 Location:{Index:0 Col:5 Row:3 file:module-0.rego text:q}}
| | | | | *ir.Block Block (6 statements)
| | | | | | *ir.IsUndefinedStmt &{Source:3 Location:{Index:0 Col:5 Row:6 file:module-0.rego text:else = true}}
| | | | | | *ir.MakeBooleanStmt &{Value:true Target:5 Location:{Index:0 Col:6 Row:7 file:module-0.rego text:true}}
| | | | | | *ir.MakeBooleanStmt &{Value:false Target:6 Location:{Index:0 Col:6 Row:7 file:module-0.rego text:true}}
| | | | | | *ir.NotEqualStmt &{A:5 B:6 Location:{Index:0 Col:6 Row:7 file:module-0.rego text:true}}
| | | | | | *ir.MakeBooleanStmt &{Value:true Target:7 Location:{Index:0 Col:5 Row:6 file:module-0.rego text:else = true}}
| | | | | | *ir.AssignVarStmt &{Source:7 Target:3 Location:{Index:0 Col:5 Row:6 file:module-0.rego text:else = true}}
| | | | | *ir.Block Block (2 statements)
| | | | | | *ir.IsDefinedStmt &{Source:3 Location:{Index:0 Col:5 Row:6 file:module-0.rego text:else = true}}
| | | | | | *ir.AssignVarOnceStmt &{Target:2 Source:3 Location:{Index:0 Col:5 Row:6 file:module-0.rego text:else = true}}
| | | *ir.Block Block (5 statements)
| | | | *ir.MakeBooleanStmt &{Value:true Target:5 Location:{Index:0 Col:9 Row:9 file:module-0.rego text:false}}
| | | | *ir.MakeBooleanStmt &{Value:false Target:6 Location:{Index:0 Col:9 Row:9 file:module-0.rego text:false}}
| | | | *ir.NotEqualStmt &{A:5 B:6 Location:{Index:0 Col:9 Row:9 file:module-0.rego text:false}}
| | | | *ir.MakeBooleanStmt &{Value:false Target:7 Location:{Index:0 Col:5 Row:9 file:module-0.rego text:q = false}}
| | | | *ir.AssignVarStmt &{Source:7 Target:4 Location:{Index:0 Col:5 Row:9 file:module-0.rego text:q = false}}
| | | *ir.Block Block (2 statements)
| | | | *ir.IsDefinedStmt &{Source:4 Location:{Index:0 Col:5 Row:9 file:module-0.rego text:q = false}}
| | | | *ir.AssignVarOnceStmt &{Target:2 Source:4 Location:{Index:0 Col:5 Row:9 file:module-0.rego text:q = false}}
| | | *ir.Block Block (1 statements)
| | | | *ir.ReturnLocalStmt &{Source:2 Location:{Index:0 Col:5 Row:3 file:module-0.rego text:q}}
```